exporter/collector/monitoredresource.go

```diff
@@ -137,28 +146,56 @@ func (m *metricMapper) resourceMetricsToMonitoredResource(
 	case semconv.AttributeCloudPlatformAWSEC2:
 		return createMonitoredResource(awsEc2Instance, attrs)
 	default:
-		// Fallback to generic_task
-		_, hasServiceName := attrs.Get(semconv.AttributeServiceName)
-		_, hasServiceInstanceID := attrs.Get(semconv.AttributeServiceInstanceID)
-		if hasServiceName && hasServiceInstanceID {
-			return createMonitoredResource(genericTask, attrs)
-		}
+		return fallbackMonitoredResource(attrs)
+	}
+}
 
-		// If not possible, fallback to generic_node
-		return createMonitoredResource(genericNode, attrs)
+func fallbackMonitoredResource(resourceAttrs pdata.AttributeMap) (*monitoredrespb.MonitoredResource, labels) {
+	// Fallback to generic_task
+	_, hasServiceName := resourceAttrs.Get(semconv.AttributeServiceName)
+	_, hasServiceInstanceID := resourceAttrs.Get(semconv.AttributeServiceInstanceID)
+	if hasServiceName && hasServiceInstanceID {
+		return createMonitoredResource(genericTask, resourceAttrs)
+	}
+	// If not possible, fallback to generic_node
+	return createMonitoredResource(genericNode, resourceAttrs)
+}
+
+func (m *metricMapper) createCustomMonitoredResource(
+	resourceAttrs pdata.AttributeMap,
+) (*monitoredrespb.MonitoredResource, labels) {
+	for monitoredResourceType, mapping := range m.customResourceMappings {
+		mrLabels, foundAll := mrLabelsForMapping(mapping, resourceAttrs)
+		if foundAll {
+			// TODO handle extra labels
+			return &monitoredrespb.MonitoredResource{
+				Type:   monitoredResourceType,
+				Labels: mrLabels,
+			}, labels{}
+		}
 	}
+	return fallbackMonitoredResource(resourceAttrs)
```

Comment:
curious what we should do if it doesn't match any of the custom mappings... I've written it to fall back to generic_task/generic_node.

Reply:
It might be better to put empty strings for the missing MR labels and let any errors surface from the CreateTimeSeries call. That's what we do for the others.

Reply:
What if someone adds multiple MR for one platform?  We can't really add the detection/fallback mechanisms like we do with k8s_container.  Which MR do we default to if it doesn't match either?